docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,755 @@
+
+#### `is_instance` checks :thumbsup:
+
+Strict mode also means it makes sense to provide an `is_instance` method on validators which effectively run
+validation then throws away the result while avoiding the (admittedly small) overhead of creating and raising
+and error or returning the validation result.
+
+### Formalised Conversion Table :thumbsup:
+
+As well as complaints about coercion, another (legitimate) complaint was inconsistency around data conversion.
+
+In pydantic V2, the following principle will govern when data should be converted in "lax mode" (`strict=False`):
+
+> If the input data has a SINGLE and INTUITIVE representation, in the field's type, AND no data is lost
+> during the conversion, then the data will be converted, Otherwise a validation error is raised.
+> There is one exception to this rule: string fields -
+> virtually all data has an intuitive representation as a string (e.g. `repr()` and `str()`), therefore
+> a custom rule is required: only `str`, `bytes` and `bytearray` are valid as inputs to string fields.
+
+Some examples of what that means in practice:
+
+| Field Type | Input                   | Single & Intuitive R. | data Loss        | Result  |
+|------------|-------------------------|-----------------------|------------------|---------|
+| `int`      | `"123"`                 | :material-check:      | :material-close: | Convert |
+| `int`      | `123.0`                 | :material-check:      | :material-close: | Convert |
+| `int`      | `123.1`                 | :material-check:      | :material-check: | Error   |
+| `date`     | `"2020-01-01"`          | :material-check:      | :material-close: | Convert |
+| `date`     | `"2020-01-01T12:00:00"` | :material-check:      | :material-check: | Error   |
+| `int`      | `b"1"`                  | :material-close:      | :material-close: | Error   |
+
+(For the last case converting `bytes` to an `int` could reasonably mean `int(bytes_data.decode())` or 
+`int.from_bytes(b'1', 'big/little')`, hence an error)
+
+In addition to the general rule, we'll provide a conversion table which defines exactly what data will be allowed 
+to which field types. See [the table below](#conversion-table) for a start on this.
+
+### Built in JSON support :thumbsup:
+
+pydantic-core can parse JSON directly into a model or output type, this both improves performance and avoids
+issue with strictness - e.g. if you have a strict model with a `datetime` field, the input must be a 
+`datetime` object, but clearly that makes no sense when parsing JSON which has no `datatime` type.
+Same with `bytes` and many other types.
+
+Pydantic V2 will therefore allow some conversion when validating JSON directly, even in strict mode 
+(e.g. `ISO8601 string -> datetime`, `str -> bytes`) even though this would not be allowed when validating
+a python object.
+
+In future direct validation of JSON will also allow:
+
+* parsing in a separate thread while starting validation in the main thread
```

Comment:
Totally possible, I think the best advantage would actually be the ability to stop parsing when you hit the wrong type, without completing all parsing then starting validation.

Since both JSON parsing and validation are pretty big, I imagine the input data would have to be massive before you saw a meaningful difference from streaming parsing vs. reading the whole string of the wire, then starting validation.